src/components/ListItem.jsx

```diff
@@ -0,0 +1,41 @@
+import { useState } from "react";
+
+const ListItem = ({ item, onEdited }) => {
+    const [showInput, setShowInput] = useState(false);
+    const returnGroceryList = () => {
+        return (
+            <li>
+                {showInput ? (
+                    <div className="flex flex-row items-center">
+                        <input
+                            type="checkbox"
+                        ></input>
+                        <input
+                            className="flex flex-row items-center"
+                            type="text"
+                            value={item.name}
+                            onChange={onEdited}
+                            onBlur={() => setShowInput(false)}
```

Comment:
Only call onEdited when the user finished editing (i.e. onBlur).

In the next PR, onEdited should be called with the full updated object (i.e. the item object modified with the new name / i Complete according to the checkbox)